chapter01/1.3 - URLify/urlify.js

```diff
@@ -1,24 +1,7 @@
-var urlify = function(str, length) {
-  // have a pointer to check from start to end
-  var strArr = str.split('');
-  var pointer = 0;
-  while (pointer < str.length) {
-    if (strArr[pointer] === ' ') {
-      // *** needs more work here, a little wierd
-      // not handling trailing spaces properly
-      for (var i = str.length - 1; i > pointer + 3; i--) {
-        strArr[i] = str[i - 2];
-      }
-      strArr[pointer] = '%';
-      strArr[pointer+1] = '2';
-      strArr[pointer+2] = '0';
-      console.log(strArr, strArr.length);
-    } 
-    pointer++;
-  }
-  // if character is a space, move remainder chars by two
-  // replace following three chars with '%20'
-  return strArr.join('');
-};
+function urlIfy(val) {
+  const removeTrails = val.trim();
+  return encodeURI(removeTrails);
```

Comment:
The purpose of that question was for you to write your own algorithm for `encodeURI`. Typically, in coding interviews, it's never a good idea to call built-in methods, unless they are not core to the question.

Reply:
Thanks, @profnandaa, I will keep that in mind.